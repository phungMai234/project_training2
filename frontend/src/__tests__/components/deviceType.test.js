import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import DeviceType from "../../components/deviceType/deviceType";

const mockStore = configureStore([]);

describe('Device Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      query: {
        data: {
          startDate: "2019/04/29",
          endDate: "2019/04/29"
        }
      },
      types: {
        data: {
          iOS: 40,
          android: 60
        },
        isFetching: false
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <DeviceType/>
      </Provider>
    );
  })
  it('should render Device with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
})