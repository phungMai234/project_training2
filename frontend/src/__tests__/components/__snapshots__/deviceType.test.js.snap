// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device Component should render Device with given state from Redux store 1`] = `
<div
  className="device-type"
>
  <label
    className="text-secondary"
  >
    Devive Type
  </label>
  <div
    className="d-flex justify-content-center"
  >
    <svg
      height={200}
      width={200}
    >
      <circle
        cx={150}
        cy={150}
        fill="#ffffff"
        r={50}
      />
      <g>
        <path
          d="M4.592425496802574e-15,-75A75,75,0,1,1,-44.0838939219355,60.676274578121046L-29.389262614623668,40.450849718747364A50,50,0,1,0,3.061616997868383e-15,-50Z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#00cc99",
              "padding": 10,
              "stroke": "transparent",
              "strokeWidth": 1,
            }
          }
          transform="translate(100, 100)"
        />
        <path
          d="M-44.0838939219355,60.676274578121046A75,75,0,0,1,-1.3777276490407723e-14,-75L-9.184850993605149e-15,-50A50,50,0,0,0,-29.389262614623668,40.450849718747364Z"
          role="presentation"
          shapeRendering="auto"
          style={
            Object {
              "fill": "#8c1aff",
              "padding": 10,
              "stroke": "transparent",
              "strokeWidth": 1,
            }
          }
          transform="translate(100, 100)"
        />
        <text
          direction="inherit"
          dx={0}
          dy={4.97}
          id="pie-labels-0"
          x={167}
          y={122}
        >
          <tspan
            dx={0}
            style={
              Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 20,
                "stroke": "transparent",
              }
            }
            textAnchor="start"
            x={167}
          >
             
          </tspan>
        </text>
        <text
          direction="inherit"
          dx={0}
          dy={4.97}
          id="pie-labels-1"
          x={33}
          y={78}
        >
          <tspan
            dx={0}
            style={
              Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 20,
                "stroke": "transparent",
              }
            }
            textAnchor="end"
            x={33}
          >
             
          </tspan>
        </text>
      </g>
    </svg>
    <div
      className="d-flex align-items-center"
    >
      <div>
        <div>
          <div
            className="title-pie"
          >
            <div
              className="box-custom ios"
            >
               
            </div>
            <label>
              iOS
            </label>
          </div>
          <label>
            40
             %
          </label>
        </div>
        <div>
          <div
            className="title-pie"
          >
            <div
              className="box-custom android"
            >
               
            </div>
            <label>
              Android
            </label>
          </div>
          <label>
            60
             %
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
